controlplane/kubeadm/controllers/helpers_test.go

```diff
@@ -341,6 +341,7 @@ func TestKubeadmControlPlaneReconciler_generateMachine(t *testing.T) {
 	g.Expect(machine.OwnerReferences).To(HaveLen(1))
 	g.Expect(machine.OwnerReferences).To(ContainElement(*metav1.NewControllerRef(kcp, controlplanev1.GroupVersion.WithKind("KubeadmControlPlane"))))
 	g.Expect(machine.Spec).To(Equal(expectedMachineSpec))
+	g.Expect(machine.GetAnnotations()[controlplanev1.MachineClonedTemplateAnnotation])
```

Comment:
I'm not sure this is asserting anything. I'd suggest:

```suggestion
	g.Expect(machine.GetAnnotations()).To(HaveKey(controlplanev1.MachineClonedTemplateAnnotation))
```